collect/collect.go

```diff
@@ -810,39 +871,14 @@ func (i *InMemCollector) send(trace *types.Trace, sendReason string) {
 		i.Logger.Info().WithFields(logFields).Logf("Trace would have been dropped, but dry run mode is enabled")
 	}
 	i.Logger.Info().WithFields(logFields).Logf("Sending trace")
-	for _, sp := range trace.GetSpans() {
-		if i.Config.GetAddRuleReasonToTrace() {
-			sp.Data["meta.refinery.reason"] = reason
-			sp.Data["meta.refinery.send_reason"] = sendReason
-			if key != "" {
-				sp.Data["meta.refinery.sample_key"] = key
-			}
-		}
-
-		// update the root span (if we have one, which we might not if the trace timed out)
-		// with the final total as of our send time
-		if i.isRootSpan(sp) {
-			if i.Config.GetAddCountsToRoot() {
-				sp.Data["meta.span_event_count"] = int64(trace.SpanEventCount())
-				sp.Data["meta.span_link_count"] = int64(trace.SpanLinkCount())
-				sp.Data["meta.span_count"] = int64(trace.SpanCount())
-				sp.Data["meta.event_count"] = int64(trace.DescendantCount())
-			} else if i.Config.GetAddSpanCountToRoot() {
-				sp.Data["meta.span_count"] = int64(trace.DescendantCount())
-			}
-		}
-
-		isDryRun := i.Config.GetIsDryRun()
-		if isDryRun {
-			sp.Data[config.DryRunFieldName] = shouldSend
-		}
-		if i.hostname != "" {
-			sp.Data["meta.refinery.local_hostname"] = i.hostname
-		}
-		mergeTraceAndSpanSampleRates(sp, trace.SampleRate(), isDryRun)
-		i.addAdditionalAttributes(sp)
-		i.Transmission.EnqueueSpan(sp)
+	i.outgoingTraces <- sendableTrace{
```

Comment:
Can we add a metric for the number of items in the `outgoingTraces` queue?

Reply:
Will do

Reply:
Added `collector_peer_queue` histogram.

Reply:
How about naming it `collector_send_queue`? I don't think this is the queue to send to peers

Reply:
I had a typo, the new metric is named `collector_outgoing_queue`